EasyLvlKatas/OutOfSpace.java

```diff
@@ -0,0 +1,15 @@
+package EasyLvlKatas;
+
+public class OutOfSpace {
+
+    public static String[] spacey(String[] array) {
+        String[] result = new String[array.length];
+        StringBuilder builder = new StringBuilder();
+        for(int i = 0; i < array.length; i++) {
+            builder.append(array[i]);
+            result[i] = builder.toString();
```

Comment:
There are two other options:
1. You can modify the existing array.
2. You can work directly with arrays without using StringBuilder. But your solution is ok also.